grammar.js

```diff
@@ -1334,6 +1341,34 @@ module.exports = grammar({
       ))
     },
 
+    macro_def: $ => {
+      const visibility = optional(
+        field('visibility', choice($.private, $.protected)),
+      )
+      const name = field('name', choice(
+        $.identifier,
+        alias($.identifier_method_call, $.identifier),
+        alias($._operator_token, $.operator),
+      ))
+      const params = seq(
+        '(',
+        field('params', optional(alias($.typeless_param_list, $.param_list))),
```

Comment:
Is there any harm in using the typed param list here? In other words, should tree-sitter accept this?
``` crystal
macro foo(bar : String)
end
```

Personally I'd be okay with that, just to avoid duplicate versions of rules. The tree-sitter grammar can accept code more liberally than the compiler. But if you have plans to use the tree-sitter grammar as part of LSP tooling, I can understand taking a more restrictive approach.

Reply:
> if you have plans to use the tree-sitter grammar as part of LSP tooling

I do, but that's still WIP. Making sure macros don't have types can be done later on if it makes the parser simpler though.